merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/ConstraintAction.java

```diff
@@ -152,30 +147,58 @@ public List<ConstraintResult> getViolations(final PlanId planId, final Optional<
       final var realProfiles = new HashMap<String, LinearProfile>();
       final var discreteProfiles = new HashMap<String, DiscreteProfile>();
 
+      // try to compile and run the constraint that were not
+      // successful and cached in the past
       for (final var entry : constraintCode.entrySet()) {
         final var constraint = entry.getValue();
         final Expression<ConstraintResult> expression;
 
         // TODO: cache these results, @JoelCourtney is this in reference to caching the output of the DSL compilation?
-        final var constraintCompilationResult = constraintsDSLCompilationService.compileConstraintsDSL(
-            plan.missionModelId,
-            Optional.of(planId),
-            Optional.of(simDatasetId),
-            constraint.definition()
-        );
+        ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult constraintCompilationResult = null;
```

Comment:
Minor: `constraintCompilationResult` can be final
```suggestion
        final ConstraintsDSLCompilationService.ConstraintsDSLCompilationResult constraintCompilationResult;
```

Reply:
I can't make this final because attempt to modify it inside a try-catch block which isn't allowed.

Reply:
If you don't initialize it to `null`, it won't be considered a modification when you initialize it in the try-catch. We do the same thing with `simDatasetId` on L56 of this file -- declare it as `final` and initialize it later in the `if-else`.

Reply:
ahhhh, I didn't see your suggestion removed the null.

Reply:
Done.